### PR TITLE
LPS-196094 [CP 2.0] Refactoring CIC epic

### PR DESCRIPTION
**Pull request by eduardo-diniz**
[Impedibug](https://liferay.atlassian.net/browse/LPS-196094)

[EPIC](https://liferay.atlassian.net/browse/LPS-159127)

### CODE REVIEW

**Comment by eduardo-diniz** [comment]
ci:test:sf

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/components/HighPriorityContacts/SetupHighPriorityContact.jsx [hunk 1]
```diff
@@ -54,66 +46,115 @@ const SetupHighPriorityContact = ({
...
-	const highPriorityContactsCategory = mapFilterToContactsCategory(filter);
+	useEffect(() => {
+		const result = async () => {
+			try {
+				const data = await getAccountRolesId(project, client);
+				setRolesId(data);
+			} catch (error) {
+				console.error(error);
+			}
+		};
```
Comment (kevenleone) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/components/HighPriorityContacts/SetupHighPriorityContact.jsx:63]: Inline

```suggestion
                getAccountRolesId(project, client)
                    .then(setRoleIds)
                    .catch(console.error)
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/components/HighPriorityContacts/SetupHighPriorityContact.jsx [hunk 2]
```diff
@@ -54,66 +46,115 @@ const SetupHighPriorityContact = ({
...
+		{data: userAccountsData},
+	] = useUserAccountsByAccountExternalReferenceCode(project?.accountKey);
+
+	const getHighPriorityContactsByFilterRAYSOURCE = async (filter) => {
+		return userAccountsData?.accountUserAccountsByExternalReferenceCode?.items
+			.filter((account) => {
+				return account?.selectedAccountSummary?.roleBriefs?.some(
+					(role) => role?.name === filter
+				);
+			})
```
Comment (kevenleone) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/components/HighPriorityContacts/SetupHighPriorityContact.jsx:80]:
```suggestion
			.filter((account) => ({
			     account?.selectedAccountSummary?.roleBriefs?.some(
					(role) => role?.name === filter
				);
			}))
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/components/HighPriorityContacts/SetupHighPriorityContact.jsx [hunk 3]
```diff
@@ -54,66 +46,115 @@ const SetupHighPriorityContact = ({
...
+		result();
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [project?.accountKey]);
+
+	const [
+		,
+		{data: userAccountsData},
+	] = useUserAccountsByAccountExternalReferenceCode(project?.accountKey);
+
+	const getHighPriorityContactsByFilterRAYSOURCE = async (filter) => {
```
Comment (kevenleone) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/components/HighPriorityContacts/SetupHighPriorityContact.jsx:74]: It doesn't seems to be a promise

```suggestion
	const getHighPriorityContactsByFilterRAYSOURCE = (filter) => {
```

also

```suggestion
	const getHighPriorityContactsByFilterRaysource = (filter) => {
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/components/HighPriorityContacts/SetupHighPriorityContact.jsx [hunk 4]
```diff
@@ -54,66 +46,115 @@ const SetupHighPriorityContact = ({
...
+			});
+	};
+
+	const highPriorityContactsCategory = mapFilterToContactsCategory(filter);
 
-		return response.items;
-	}
+	getHighPriorityContactsByFilterRAYSOURCE(
+		highPriorityContactsCategory.contactsCategory.role
+	);
```
Comment (kevenleone) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/components/HighPriorityContacts/SetupHighPriorityContact.jsx:113]: Why this is here??

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/components/HighPriorityContacts/SetupHighPriorityContact.jsx [hunk 5]
```diff
@@ -54,66 +46,115 @@ const SetupHighPriorityContact = ({
...
+						return {
+							email: contact?.email,
+							filter: contact.role,
+							filterId: contact.roleId,
+							id: contact?.id,
+							label: contact?.name,
+							value: (index + 1).toString(),
+						};
+					}
+				);
```
Comment (kevenleone) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/components/HighPriorityContacts/SetupHighPriorityContact.jsx:132]:
```suggestion
				const highPriorityContacts = getHighPriorityContactsByFilterRAYSOURCE(
					highPriorityContactsCategory.contactsCategory.role
				);
				
				const contacts = highPriorityContacts.map(
					(highPriorityContact, index) => ({
						email: highPriorityContact?.email,
						filter: highPriorityContact.role,
						filterId: highPriorityContact.roleId,
						id: highPriorityContact?.id,
						label: highPriorityContact?.name,
						value: (index + 1).toString(),
					})
				);
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/containers/setup-forms/SetupAnalyticsCloudForm/index.jsx [hunk 6]
```diff
@@ -243,49 +243,52 @@ const SetupAnalyticsCloudPage = ({
...
-										type: 'liferay-rest',
-									},
-									mutation: addIncidentReportAnalyticsCloud,
-									variables: {
-										IncidentReportContactAnalyticsCloud: {
-											analyticsCloudWorkspaceId,
-											emailAddress: email,
-											r_accountEntryToIncidentReportContactAC_accountEntryId:
-												project.id,
+					if (!featureFlags.includes('LPS-159127')) {
```
Comment (kevenleone) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/containers/setup-forms/SetupAnalyticsCloudForm/index.jsx:246]: Can you do 

if (featureFlags.includes('LPS-159127')) {
 /// blablabla
} else {
 /// blablabla
}

Keep attention in the order

I think is better than

if (!featureFlags.includes('LPS-159127'))

and

if (featureFlags.includes('LPS-159127')) // line 270

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/components/HighPriorityContacts/SetupHighPriorityContact.jsx (hunk 2 above)
Comment (eduardo-diniz) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/components/HighPriorityContacts/SetupHighPriorityContact.jsx:80]: if I removed this return, the function not working.

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/containers/setup-forms/SetupAnalyticsCloudForm/index.jsx (hunk 6 above)
Comment (eduardo-diniz) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/common/containers/setup-forms/SetupAnalyticsCloudForm/index.jsx:246]: Thanks for suggestion

**Comment by eduardo-diniz** [comment]
ci:test:sf

**Comment by kevenleone** [comment]
Just starting reviewing :)

**Comment by eduardo-diniz** [comment]
ci:test:sf

**Comment by eduardo-diniz** [comment]
ci:test:relevant

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/ActivationStatus/LiferayExperienceCloud/components/SetupLXCModal/components/SetupLXCForm/components/SetupLXCPage/index.jsx [hunk 7]
```diff
@@ -68,13 +64,16 @@ const SetupLiferayExperienceCloudPage = ({
 	};
 
 	const addHighPriorityContacts = (contactList) => {
-		setAddHighPriorityContactList((oldList) => {
+		setAddHighPriorityContact((oldList) => {
```
Comment (kevenleone) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/ActivationStatus/LiferayExperienceCloud/components/SetupLXCModal/components/SetupLXCForm/components/SetupLXCPage/index.jsx:67]: Don't use:
1. old
2. List

In this case, we use `prev` and the name of state, in this case highPriorityContact

```suggestion
		setAddHighPriorityContact((prevHighPriorityContacts) => {
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/ActivationStatus/LiferayExperienceCloud/components/SetupLXCModal/components/SetupLXCForm/components/SetupLXCPage/index.jsx [hunk 8]
```diff
@@ -68,13 +64,16 @@ const SetupLiferayExperienceCloudPage = ({
...
 			const uniqueContacts = [
 				...oldList,
 				...contactList.filter(
 					(contact) =>
 						!oldList.some(
-							(oldContact) => oldContact.id === contact.id
+							(oldContact) =>
```
Comment (kevenleone) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/ActivationStatus/LiferayExperienceCloud/components/SetupLXCModal/components/SetupLXCForm/components/SetupLXCPage/index.jsx:73]:
```suggestion
							(prevContact) =>
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/ActivationStatus/LiferayExperienceCloud/components/SetupLXCModal/components/SetupLXCForm/components/SetupLXCPage/index.jsx [hunk 9]
```diff
@@ -68,13 +64,16 @@ const SetupLiferayExperienceCloudPage = ({
...
+								oldContact.category?.role ===
+									contact.category?.role &&
+								oldContact?.id === contact?.id
```
Comment (kevenleone) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/ActivationStatus/LiferayExperienceCloud/components/SetupLXCModal/components/SetupLXCForm/components/SetupLXCPage/index.jsx:76]:
```suggestion
								prevContact.category?.role ===
									contact.category?.role &&
								prevContact?.id === contact?.id
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/ActivationStatus/LiferayExperienceCloud/components/SetupLXCModal/components/SetupLXCForm/components/SetupLXCPage/index.jsx [hunk 10]
```diff
@@ -84,13 +83,16 @@ const SetupLiferayExperienceCloudPage = ({
 	};
 
 	const removeHighPriorityContacts = (contactList) => {
-		setRemoveHighPriorityContactList((oldList) => {
+		setRemoveHighPriorityContact((oldList) => {
```
Comment (kevenleone) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/ActivationStatus/LiferayExperienceCloud/components/SetupLXCModal/components/SetupLXCForm/components/SetupLXCPage/index.jsx:86]:
```suggestion
		setRemoveHighPriorityContact((prevHighPriorityContact) => {
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/ActivationStatus/LiferayExperienceCloud/components/SetupLXCModal/components/SetupLXCForm/components/SetupLXCPage/index.jsx [hunk 11]
```diff
@@ -84,13 +83,16 @@ const SetupLiferayExperienceCloudPage = ({
...
 			const uniqueContacts = [
 				...oldList,
 				...contactList.filter(
 					(contact) =>
 						!oldList.some(
-							(oldContact) => oldContact.id === contact.id
+							(oldContact) =>
+								oldContact.category?.role ===
+									contact.category?.role &&
+								oldContact?.id === contact?.id
```
Comment (kevenleone) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/components/ActivationStatus/LiferayExperienceCloud/components/SetupLXCModal/components/SetupLXCForm/components/SetupLXCPage/index.jsx:95]:
```suggestion
							(prevContact) =>
								prevContact.category?.role ===
									contact.category?.role &&
								prevContact?.id === contact?.id
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/pages/Project/TeamMembers/components/TeamMembersTable/TeamMembersTable.jsx [hunk 12]
```diff
@@ -110,79 +109,49 @@ const TeamMembersTable = ({
...
+					(role) => role?.name === filter
+				);
+			})
+			.map((account) => {
+				const {emailAddress} = account;
+
+				return {
+					email: emailAddress,
+				};
+			});
```
Comment (kevenleone) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/pages/Project/TeamMembers/components/TeamMembersTable/TeamMembersTable.jsx:125]: Simplify

```suggestion
		return userAccountsData?.accountUserAccountsByExternalReferenceCode?.items
			.filter((account) => account?.selectedAccountSummary?.roleBriefs?.some(
					(role) => role?.name === filter
				))
			.map((account) => ({
					email: account.emailAddress,
				});
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/pages/Project/TeamMembers/components/TeamMembersTable/TeamMembersTable.jsx [hunk 13]
```diff
@@ -110,79 +109,49 @@ const TeamMembersTable = ({
...
 	};
 
 	useEffect(() => {
 		const fetchHighPriorityContacts = async () => {
+			const highPriorityContactsPromises = rolesHighPriorityContacts.map(
+				(role) => getHighPriorityContactsByFilter(role)
+			);
```
Comment (kevenleone) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/pages/Project/TeamMembers/components/TeamMembersTable/TeamMembersTable.jsx:132]: Remove variable

```suggestion

```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/pages/Project/TeamMembers/components/TeamMembersTable/TeamMembersTable.jsx [hunk 14]
```diff
@@ -110,79 +109,49 @@ const TeamMembersTable = ({
...
-				for (const filter of Object.keys(
-					HIGH_PRIORITY_CONTACT_CATEGORIES
-				)) {
-					updatedFilteredContacts[
-						filter
-					] = await getHighPriorityContactsByFilter(filter);
-				}
+				const highPriorityContactsResults = await Promise.all(
+					highPriorityContactsPromises
+				);
```
Comment (kevenleone) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/pages/Project/TeamMembers/components/TeamMembersTable/TeamMembersTable.jsx:137]:
```suggestion
				const highPriorityContactsResults = await Promise.all(
					rolesHighPriorityContacts.map(
 				               (role) => getHighPriorityContactsByFilter(role)
 			                 )
				);
```

Comment (kevenleone) [reply modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/pages/Project/TeamMembers/components/TeamMembersTable/TeamMembersTable.jsx:137]: Inline

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/utils/getHighPriorityContacts.js [hunk 15]
```diff
@@ -19,7 +20,42 @@ const HIGH_PRIORITY_CONTACT_CATEGORIES = {
...
+	if (filter.includes('privacy')) {
+		return 'Data Breach Contact';
+	}
+
+	if (filter.includes('security')) {
+		return 'Security Incident Contact';
+	}
+	if (filter.includes('critical')) {
+		return 'Critical Incident Contact';
+	}
```
Comment (kevenleone) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/utils/getHighPriorityContacts.js:33]: Space

```suggestion
	if (filter.includes('security')) {
		return 'Security Incident Contact';
	}
	
	if (filter.includes('critical')) {
		return 'Critical Incident Contact';
	}
```

File: modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/utils/getHighPriorityContacts.js [hunk 16]
```diff
@@ -19,7 +20,42 @@ const HIGH_PRIORITY_CONTACT_CATEGORIES = {
...
+};
+
+const rolesHighPriorityContacts = [
+	'Data Breach Contact',
+	'Security Incident Contact',
+	'Critical Incident Contact',
+];
```
Comment (kevenleone) [inline modules/dxp/apps/osb/osb-site-initializer/osb-site-initializer-customer-portal/extra/remote-app/src/routes/customer-portal/utils/getHighPriorityContacts.js:40]: Should we use i18n here?

**Review by kevenleone — CHANGES_REQUESTED** [review]
@eduardo-diniz GOOD GOOD GOOD job on that Eduardo. 

The code looks amazing, congratulations 🎉 

I left few comments, just to simplify or rename few things here

**Comment by eduardo-diniz** [comment]
ci:test:relevant

**Comment by kevenleone** [comment]
Just starting reviewing :)

**Comment by kevenleone** [comment]
resent: https://github.com/brianchandotcom/liferay-portal/pull/141342

**Pull request closed without merging**